### PR TITLE
Update flags/args of init cmd

### PR DESCRIPTION
**Pull request by daved**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1871" title="DX-1871" target="_blank">TaskDX-1871</a>  `state init` inconsistent with `state checkout`
  </td></table>
  <br />

### CODE REVIEW

File: cmd/state/internal/cmdtree/init.go [hunk 1]
```diff
@@ -22,22 +22,17 @@ func newInitCommand(prime *primer.Values) *captain.Command {
...
 				Name:   "skeleton",
 				Value:  p.StrP(""),
 				Hidden: true,
 			},
 			{
-				// Hidden flag for legacy Komodo support
-				Name:   "language",
-				Value:  &params.Language,
-				Hidden: true,
+				// Supports Komodo
```
Comment (Naatan) [inline cmd/state/internal/cmdtree/init.go:32]:
```suggestion
```

File: internal/locale/locales/en-us.yaml [hunk 2]
```diff
@@ -1246,12 +1246,12 @@ arg_state_init_namespace:
...
-arg_state_init_language_description:
-  other: The language that this project should use, and optionally the version of that language
-flag_state_init_path_description:
-  other: Where to initialize the project
+flag_state_init_language_description:
+  other: "The Language[@version] that this project should use"
+arg_state_init_path:
+  other: path
+arg_state_init_path_description:
+  other: "Where to initialize the project (Default: Current working directory)"
```
Comment (Naatan) [inline internal/locale/locales/en-us.yaml:1254]:
```suggestion
  other: "Where to initialize the project"
```

We don't seem to do this for other commands. Plus doing this will lead to bugs when we change the default behavior but forget the description cause it's not covered by our type system.

File: internal/output/progress_win.go [hunk 3]
```diff
@@ -47,7 +47,7 @@ func (d *Spinner) moveCaretBackInCommandPrompt(n int) {
 		cursor.x = csbi.cursorPosition.x + short(-n)
 		cursor.y = csbi.cursorPosition.y
 
-		r2, _, err2 := procSetConsoleCursorPosition.Call(uintptr(handle), uintptr(*(*int32)(unsafe.Pointer(&cursor))))
+		r2, _, err2 := procSetConsoleCursorPosition.Call(uintptr(handle), uintptr(*((*uint32)(unsafe.Pointer(&cursor)))))
```
Comment (Naatan) [inline internal/output/progress_win.go:50]: This seems unrelated to this PR?

**Review by Naatan — CHANGES_REQUESTED** [review]

File: internal/output/progress_win.go (hunk 3 above)
Comment (mitchell-as) [reply internal/output/progress_win.go:50]: Looks like he pulled in this fix from v0.40.

**Review by Naatan — APPROVED** [review]
